### ISSUE TITLE
Support Flatcar as Guest OS

### ISSUE THREAD

**Issue by dongsupark** [vmware/govmomi#2088]
I have been working on supporting [Flatcar Container Linux](https://www.flatcar-linux.org/) in [image-builder](https://github.com/kubernetes-sigs/image-builder).

While trying out [Packer](https://github.com/hashicorp/packer) for vmware-iso as well as vsphere-iso, I found out that govmomi defines a variable [VirtualMachineGuestOsIdentifierCoreos64Guest](https://github.com/vmware/govmomi/blob/609218c5232673cec215d088eb01d043120de385/vim25/types/enum.go#L4399), but not the equivalent for Flatcar.

So basically I would like to add another variable for Flatcar, like `VirtualMachineGuestOsIdentifierFlatcar64Guest`.

Though I am wondering if it is simply possible to do so, or if there are things to do in other places, like API or open-vm-tools.

**Comment by dougm** [comment]
The IDs are defined by the vSphere API, not govmomi: https://code.vmware.com/apis/358/vsphere/doc/vim.vm.GuestOsDescriptor.GuestOsIdentifier.html

You can use `otherLinux64Guest` for example in this case.

**Comment by dongsupark** [comment]
Thanks for the reply.
Yeah, I am aware of other IDs like `otherLinux64Guest`.

Ultimately what we would like to do is to add `flatcar64Guest` just like `coreos64Guest`, into the [doc](https://code.vmware.com/apis/358/vsphere/doc/vim.vm.GuestOsDescriptor.GuestOsIdentifier.html).
Is there a public github repo where we can add such an ID via pull request?

**Comment by jakerobinson** [comment]
I’ll send this link to the PM!

**Comment by jakerobinson** [comment]
The feature request has been submitted to the Guest OS team. I'll update if I hear anything!

**Comment by dougm** [comment]
Thanks @jakerobinson . Closing this issue, if/when flatcar support comes in, govmomi will get the new guest ID as part of an API update.

**Comment by jsavanyo4vmw** [comment]
@dongsupark If we introduced a new Guest ID for Flatcar, then only new future versions for ESXi would be aware of such an ID. Attempting to use this ID on older versions of ESXi would result in errors. So I would advise that you use Other 3x Linux 64 selection. See https://github.com/flatcar-linux/Flatcar/issues/154

**Comment by dongsupark** [comment]
Thanks!
